pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -556,62 +556,61 @@ public ResultSet execSQLQuery(String s) throws SQLException {
   @Override
   public ResultSet execSQLQuery(String s, int resultSetType, int resultSetConcurrency)
       throws SQLException {
-    BaseStatement stat = (BaseStatement) createStatement(resultSetType, resultSetConcurrency);
-    boolean hasResultSet = stat.executeWithFlags(s, QueryExecutor.QUERY_SUPPRESS_BEGIN);
+    try (BaseStatement stat = (BaseStatement) createStatement(resultSetType, resultSetConcurrency)) {
+      boolean hasResultSet = stat.executeWithFlags(s, QueryExecutor.QUERY_SUPPRESS_BEGIN);
 
-    while (!hasResultSet && stat.getUpdateCount() != -1) {
-      hasResultSet = stat.getMoreResults();
-    }
+      while (!hasResultSet && stat.getUpdateCount() != -1) {
+        hasResultSet = stat.getMoreResults();
+      }
 
-    if (!hasResultSet) {
-      throw new PSQLException(GT.tr("No results were returned by the query."), PSQLState.NO_DATA);
-    }
+      if (!hasResultSet) {
+        throw new PSQLException(GT.tr("No results were returned by the query."), PSQLState.NO_DATA);
+      }
 
-    // Transfer warnings to the connection, since the user never
-    // has a chance to see the statement itself.
-    SQLWarning warnings = stat.getWarnings();
-    if (warnings != null) {
-      addWarning(warnings);
-    }
+      // Transfer warnings to the connection, since the user never
+      // has a chance to see the statement itself.
+      SQLWarning warnings = stat.getWarnings();
+      if (warnings != null) {
+        addWarning(warnings);
+      }
 
-    return castNonNull(stat.getResultSet(), "hasResultSet==true, yet getResultSet()==null");
+      return castNonNull(stat.getResultSet(), "hasResultSet==true, yet getResultSet()==null");
+    }
   }
 
   @Override
   public void execSQLUpdate(String s) throws SQLException {
-    BaseStatement stmt = (BaseStatement) createStatement();
-    if (stmt.executeWithFlags(s, QueryExecutor.QUERY_NO_METADATA | QueryExecutor.QUERY_NO_RESULTS
-        | QueryExecutor.QUERY_SUPPRESS_BEGIN)) {
-      throw new PSQLException(GT.tr("A result was returned when none was expected."),
-          PSQLState.TOO_MANY_RESULTS);
-    }
+    try (BaseStatement stmt = (BaseStatement) createStatement()) {
```

Comment:
But this is actually an improvement in case of an exception.

Reply:
agreed